www/src/templates/template-blog-post.js

```diff
@@ -24,118 +46,272 @@ const BlogPostTemplate = React.createClass({
+        src: post.frontmatter.image.childImageSharp.responsiveSizes.src,
+        srcSet: post.frontmatter.image.childImageSharp.responsiveSizes.srcSet,
+        className: `gatsby-resp-image-image`,
+        css: {
+          width: `100%`,
+          margin: 0,
+          verticalAlign: `middle`,
+          position: `absolute`,
+          boxShadow: `inset 0px 0px 0px 400px #fff`,
+        },
+        sizes: post.frontmatter.image.childImageSharp.responsiveSizes.sizes,
+      }
+      if (post.frontmatter.imageTitle) {
+        imageProps.alt = post.frontmatter.imageTitle
+      }
+    }
     return (
-      <Container>
-        {/* Add long list of social meta tags */}
-        <Helmet>
-          <title>{post.frontmatter.title}</title>
-          <link
-            rel="author"
-            href={`https://gatsbyjs.org${post.frontmatter.author.slug}`}
-          />
-          <meta
-            name="description"
-            content={
-              post.frontmatter.excerpt ? post.frontmatter.excerpt : post.excerpt
-            }
-          />
-          <meta name="og:description" content={post.excerpt} />
-          <meta name="twitter:description" content={post.excerpt} />
-          <meta name="og:title" content={post.frontmatter.title} />
-          <meta
-            name="og:image"
-            content={post.frontmatter.image.childImageSharp.resize.src}
-          />
-          <meta
-            name="twitter:image"
-            content={post.frontmatter.image.childImageSharp.resize.src}
-          />
-          <meta name="og:type" content="article" />
-          <meta name="article:author" content={post.frontmatter.author.id} />
-          <meta
-            name="twitter:creator"
-            content={post.frontmatter.author.twitter}
-          />
-          <meta name="author" content={post.frontmatter.author.id} />
-          <meta name="twitter:label1" content="Reading time" />
-          <meta name="twitter:data1" content={`${post.timeToRead} min read`} />
-          <meta
-            name="article:published_time"
-            content={post.frontmatter.rawDate}
-          />
-        </Helmet>
-        <header
-          css={{
-            display: `flex`,
-            marginTop: rhythm(-1 / 4),
-            marginBottom: rhythm(1),
-            [presets.Tablet]: {
-              marginTop: rhythm(1 / 2),
-              marginBottom: rhythm(2),
-            },
-          }}
-        >
-          <div
+      <div>
+        <Container className="post" css={{ paddingBottom: `0 !important` }}>
+          {/* Add long list of social meta tags */}
+          <Helmet>
+            <title>
+              {post.frontmatter.title}
+            </title>
+            <link
+              rel="author"
+              href={`https://gatsbyjs.org${post.frontmatter.author.slug}`}
+            />
+            {post.frontmatter.canonicalLink &&
+              <link rel="canonical" href={post.frontmatter.canonicalLink} />}
```

Comment:
One of the things this horrible PR does is based on the presence of `canonicalLink` and `publishedAt in` frontmatter, output "Originally posted at ${publishedAt}" with the href pointing to `canonicalLink` in the blog post header.

I thought it'd be smart to use that information to output a `<link rel="canonical">` and added that along the way. Then through inspecting the source, I found out about https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-plugin-canonical-urls …

* Should I remove the additional ?
* camelCase is OK for fields in frontmatter?

Reply:
It definitely makes sense to have the canonical link point to the other site. But this way would add two links which is confusing :-)

Perhaps the other plugin could expose an API for overriding the default canonical link or something.

But another excellent idea is to just drop it in interest of making it easier to land this PR :-)

Reply:
;) :D – done!